website_mrp_bom/controllers/main.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2018 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import http
+from odoo.http import request
+
+
+class WebsiteMrpBomPage(http.Controller):
+
+    @http.route(['/components/<model("mrp.bom"):bom>'], type='http',
+                auth="public", website=True)
+    def boms_detail(self, bom, **post):
+        if isinstance(bom, request.env['mrp.bom'].__class__):
+            is_website_publisher = request.env['res.users'].has_group(
```

Comment:
I'd split this to a specific method to ease override of permission override

Reply:
I've made some changes, but I don't find way to split it instead of adding more functions.